src/app.jsx

```diff
@@ -20,33 +20,53 @@
 import cockpit from 'cockpit';
 import React, { useEffect, useState } from 'react';
 import { Button, Card, CardBody, CardTitle, CardHeader, Dropdown, DropdownItem, DropdownList, Flex, FlexItem, Icon, MenuToggle, Page, PageBreadcrumb, PageSection, SearchInput } from "@patternfly/react-core";
-import { ArrowLeftIcon, ArrowRightIcon, EllipsisVIcon, FileIcon, FolderIcon, ListIcon, PficonHistoryIcon } from "@patternfly/react-icons";
+import { ArrowLeftIcon, ArrowRightIcon, EllipsisVIcon, FileIcon, FolderIcon, ListIcon } from "@patternfly/react-icons";
 
 const _ = cockpit.gettext;
 
-// TODO: refresh automatically when files change
 export const Application = () => {
-    const [currentUser, setCurrentUser] = useState("");
     const [currentFilter, setCurrentFilter] = useState("");
     const [files, setFiles] = useState([]);
     const [path, setPath] = useState([]);
     const [pathIndex, setPathIndex] = useState(0);
 
     const onFilterChange = (_, value) => setCurrentFilter(value);
 
-    // TODO: Start navigator on user home directory
     useEffect(() => {
         cockpit.user().then(user => {
-            setCurrentUser(user.name || "");
-        })
-                .then(() => {
-                    const currentPath = path.slice(0, pathIndex).join("/");
-                    cockpit.spawn(["ls", "-p", `/${currentPath}`], { superuser: true }).then((res) => {
-                        res = res.split("\n");
-                        setFiles(res.slice(0, res.length - 1));
-                    });
-                });
-    }, [currentUser, path, pathIndex]);
+            const userPath = user.home.split("/").slice(1);
+            setPath(userPath);
+            setPathIndex(userPath.length);
+        });
+    }, []);
+
+    useEffect(() => {
+        const result = [];
+        const currentPath = path.slice(0, pathIndex).join("/");
+        const channel = cockpit.channel({
+            payload: "fslist1",
+            path: `/${currentPath}`,
+            superuser: "try",
+            watch: true,
+        });
+
+        channel.addEventListener("ready", () => {
+            setFiles(result);
+        });
+
+        channel.addEventListener("message", (ev, data) => {
+            const item = JSON.parse(data);
+            if (item && item.path && item.event === 'present' && item.path[0] !== ".")
```

Comment:
It would be really weird if `item` was undefined here. So you can drop the first two conditions. For the last one you can use `item.path.?[0]`, but I don't think it's necessary. It already checked the event type, so you can rely on .path being present.

I assume that last check is for ignoring hidden files (starting with a dot)? It's of course okay to do that at this early stage, but perhaps there should be an issue about adding an option to show dotfiles?

Reply:
Hmm, I had copied the conditions from https://github.com/cockpit-project/cockpit/blob/main/pkg/lib/cockpit-components-file-autocomplete.jsx#L116, but I can drop them here :+1: 

Yup, we haven't decided on how to handle hidden files yet so I just hide them for now.